lib/rubocop/cop/chef/correctness/invalid_platform_helper.rb

```diff
@@ -32,18 +32,18 @@ module ChefCorrectness
         #   platform?('redhat)
         #   platform?('suse')
         class InvalidPlatformHelper < Cop
-          MSG = 'Pass valid platforms to the platform? helper.'.freeze
+          include ::RuboCop::Chef::PlatformHelpers
 
-          INVALID_PLATFORMS = %w(darwin sles rhel macos mac_os).freeze
+          MSG = 'Pass valid platforms to the platform? helper.'.freeze
 
-          def_node_matcher :incorrect_platform?, <<-PATTERN
+          def_node_matcher :platform_helper?, <<-PATTERN
             (send nil? :platform? $str*)
           PATTERN
 
           def on_send(node)
-            incorrect_platform?(node) do |plat|
+            platform_helper?(node) do |plat|
               plat.to_a.each do |p|
-                add_offense(p, location: :expression, message: MSG, severity: :refactor) if INVALID_PLATFORMS.include?(p.value)
+                add_offense(p, location: :expression, message: MSG, severity: :refactor) if INVALID_PLATFORMS.key?(p.value)
```

Comment:
Would it be better to reverse the check by having a list of VALID_PLATFORMS, and reporting if the mentioned platform isn't in the list? It seems like the number of invalids is pretty unlimited, but the number of valid is known.

Reply:
Anything that comes back from /etc/os-release is a valid platform so we don't have a list of valids. We can only tell if something is known to be invalid based on common typos. It's unfortuante, but that's what we have to work with.